src/mongo/db/mongod_main.cpp

```diff
@@ -909,7 +909,8 @@ Status shutdownProcessByDBPathPidFile(const std::string& dbpath) {
                 str::stream() << "Failed to kill process: " << errnoWithDescription(e)};
     }
 
-    while (boost::filesystem::exists(pidfile)) {
+    uintmax_t size;
+    while ((size = boost::filesystem::file_size(pidfile)) != 0 && size != static_cast<uintmax_t>(-1)) {
```

Comment:
I wanted to just use `boost::filesystem::is_empty` here, but it's implemented as `file_size(...) == 0`, and https://www.boost.org/doc/libs/1_68_0/libs/filesystem/doc/reference.html#file_size says that `file_size` will return `-1` casted to be an unsigned integer when the file doesn't exist or isn't a regular file, so I ended up with this blurb (to avoid the code calling `stat` more than once per loop unnecessarily).